spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -0,0 +1,220 @@
+
+import org.apache.commons.logging.Log;
+
+import org.springframework.amqp.AmqpRejectAndDontRequeueException;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.amqp.rabbit.connection.Connection;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.connection.RabbitUtils;
+import org.springframework.amqp.rabbit.support.DefaultMessagePropertiesConverter;
+import org.springframework.amqp.rabbit.support.MessagePropertiesConverter;
+import org.springframework.core.task.SimpleAsyncTaskExecutor;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.util.Assert;
+
+import com.rabbitmq.client.AMQP.BasicProperties;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.DefaultConsumer;
+import com.rabbitmq.client.Envelope;
+
+/**
+ * The {@code SimpleMessageListenerContainer} is not so simple. Recent changes to the
+ * rabbitmq java client has facilitated a much simpler listener container that invokes the
+ * listener directly on the rabbit client consumer thread. There is no txSize property -
+ * each message is acked (or nacked) individually.
+ * @author Gary Russell
+ * @since 2.0
+ *
+ */
+public class DirectMessageListenerContainer extends AbstractMessageListenerContainer {
+
+	private final List<SimpleConsumer> consumers = new ArrayList<SimpleConsumer>();
+
+	private final MessagePropertiesConverter messagePropertiesConverter = new DefaultMessagePropertiesConverter();
+
+	private volatile TaskExecutor taskExecutor;
+
+	private volatile int consumersPerQueue = 1;
+
+	private volatile int prefetch = 1;
+
+	private volatile boolean defaultRequeueRejected = false;
+
+	private volatile long recoveryInterval;
+
+	/**
+	 * Create an instance with the provided connection factory.
+	 * @param connectionFactory the connection factory.
+	 */
+	public DirectMessageListenerContainer(ConnectionFactory connectionFactory) {
+		setConnectionFactory(connectionFactory);
+	}
+
+	/**
+	 * Set a task executor for the container - used to create the consumers not at
+	 * runtime.
+	 * @param taskExecutor the task excutor.
+	 */
+	public void setTaskExecutor(TaskExecutor taskExecutor) {
+		this.taskExecutor = taskExecutor;
+	}
+
+	/**
+	 * Each queue runs in its own consumer; set this property to create multiple
+	 * consumers for each queue.
+	 * @param consumersPerQueue the consumers per queue.
+	 */
+	public void setConsumersPerQueue(int consumersPerQueue) {
+		this.consumersPerQueue = consumersPerQueue;
+	}
+
+	/**
+	 * Set to false to requeue failed deliveries.
+	 * @param requeueRejected the requeue rejected.
+	 */
+	public void setDefaultRequeueRejected(boolean requeueRejected) {
+		this.defaultRequeueRejected = requeueRejected;
+	}
+
+	@Override
+	protected void doInitialize() throws Exception {
+		Assert.notNull(getQueueNames(), "queue(s) are required");
```

Comment:
I think according https://jira.spring.io/browse/AMQP-637 requirement we even can start container without queues.
Am I missing anything?